### PR TITLE
Verify client and combiner token at reducer

### PR DESCRIPTION
**Pull request by Wrede**
## Status

- [x] Ready
- [ ] Draft
- [ ] Hold

## Description
Make use of tokens being sent in request (header). Will only check against a single token.  

## Types of changes

What types of changes does your code introduce to FEDn?

- [ ] Hotfix (fixing a critical bug in production)
- [ ] Bugfix
- [x] New feature
- [ ] Documentation update

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [ ] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [ ] I have read the [CONTRIBUTING](https://github.com/scaleoutsystems/fedn/blob/master/CONTRIBUTING.md) doc
- [ ] I have included tests to complement my changes
- [ ] I have updated the related documentation (if necessary) 
- [ ] I have added a reviewer for this pull request
- [ ] I have added myself as an author for this pull request

## Further comments

Anything else you think we should know before merging your code!

### CODE REVIEW

File: fedn/fedn/clients/reducer/restservice.py [hunk 1]
```diff
@@ -23,6 +24,31 @@
...
+    """
+    
+    if not 'Authorization' in r.headers:
+        print("Authorization failed, missing in the header of the request", flush=True)
+        abort(401) #Unauthorized response
+    try:
+        request_token = r.headers.get('Authorization')
+        request_token = request_token.split()[1] # str: 'Token {}'.format(token)
+        if request_token == token:
+            print("Authorization accepted via token", flush=True)
```
Comment (ahellander) [inline fedn/fedn/clients/reducer/restservice.py:43]: Don't print on success.

**Review by ahellander — CHANGES_REQUESTED** [review]
Looks solid, just a minor comment.

**Review by ahellander — CHANGES_REQUESTED** [review]
Great work! 

When a combiner tries to connect with an invalid token, it still stays in the retry loop, but there is no way to recover from this error without exiting and setting the correct token. So the combiner should exit when this happens. 

Same for the client.

**Comment by Wrede** [comment]
> 

Done!

**Review by ahellander — APPROVED** [review]

**Pull request merged**